libcontainer/container_linux.go

```diff
@@ -873,6 +873,26 @@ func (c *Container) newInitConfig(process *Process) *initConfig {
 func (c *Container) Destroy() error {
 	c.m.Lock()
 	defer c.m.Unlock()
+	if !c.config.Namespaces.IsPrivate(configs.NEWPID) {
```

Comment:
I am very much against doing this. `Kill` should do all the killing, and `Destroy` should remove the files. This was all mixed up before, but I got it untangled in #3825 (alas, with a couple of regressions which you have reported in #4047).

The alternative to doing this, without mixing the kill and destroy again, is this: https://github.com/opencontainers/runc/pull/4102/commits/7de61c4d5ae19b2020c925a0c46c9e3ffd5bc6be (PTAL 🙏🏻)

Reply:
But I think we can't force users to use `runc kill` before `runc delete` if the container is in `stopped` state.